rooms.js

```diff
@@ -1242,6 +1247,20 @@ class ChatRoom extends BasicRoom {
 		this.chatRoomData = (options.isPersonal ? null : options);
 		Object.assign(this, options);
 		if (this.auth) Object.setPrototypeOf(this.auth, null);
+		/** @type {Room?} */
+		this.parent = null;
+		if (this.parent) {
+			/** @type {Room} */
+			let main = Rooms(this.parent);
+
+			if (main && main.subRooms) {
+				main.subRooms.set(this.id, this);
+				this.parent = main;
+			}
+		}
+
+		/** @type {Map<string, ChatRoom>} */
+		this.subRooms = new Map();
```

Comment:
Should probably not exist if the room has no subrooms.

Reply:
I check for ``this.subRooms.size`` in places. I don't think the memory usage matters, and it makes it easier to add subrooms.

Reply:
The parent signature is `@type {Map<string, ChatRoom>?}` so you should be checking for map existence anyway.